src/pages/new.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { NextPage } from 'next'
+import { useSingleStringUrlParameter } from '../hooks/common/use-single-string-url-parameter'
+import { useAsync } from 'react-use'
+import { createNote } from '../api/notes'
+import { AsyncLoadingBoundary } from '../components/common/async-loading-boundary'
+import { Redirect } from '../components/common/redirect'
+
+/**
+ * Creates a new note, optionally including the passed content and redirects to that note.
+ */
+export const NewNotePage: NextPage = () => {
+  const newContent = useSingleStringUrlParameter('content', '')
+
+  const { loading, error, value } = useAsync(() => {
+    return createNote(newContent)
+  }, [newContent])
+
+  return (
+    <AsyncLoadingBoundary loading={loading} componentName={'NewNotePage'} error={error}>
```

Comment:
If (for whatever reason) the creation fails then you just get an alert with "An unexpected error occurred while loading 'NewNotePage'.".
What about extending the boundary with a new optional prop that defines an alternative i18n key?

Reply:
Changed